ganttproject/src/main/java/biz/ganttproject/ganttview/TaskFilterAction.kt

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 BarD Software s.r.o
+
+This file is part of GanttProject, an open-source project management tool.
+
+GanttProject is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+GanttProject is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package biz.ganttproject.ganttview
+
+import biz.ganttproject.core.option.DefaultBooleanOption
+import net.sourceforge.ganttproject.action.GPAction
+import net.sourceforge.ganttproject.task.event.TaskListenerAdapter
+import net.sourceforge.ganttproject.task.event.TaskPropertyEvent
+import java.awt.event.ActionEvent
+
+/**
+ * @author dbarashev@bardsoftware.com
+ */
+internal class TaskFilterAction(
+  actionName: String,
+  private val filterManager: TaskFilterManager,
+  private val taskFilterOption: DefaultBooleanOption,
+  private val taskFilter: TaskFilter
+) : GPAction(actionName) {
+
+
+  private val taskListener = TaskListenerAdapter().also {
+    it.taskProgressChangedHandler = { e: TaskPropertyEvent ->
```

Comment:
This class looks generic, why there is a listener specifically on the completion percentage? Copy'n'paste from FilterCompletedTasks? In the latter case, any sense to remove the duplicated code?